components/search-form.tsx

```diff
@@ -26,36 +28,54 @@ export function SearchForm({
   filterItems,
   sortSettings,
   autoSearch,
+  initialValue,
 }: {
   itemName: string;
   onSubmitKeywords: (keywords: string, filters?: any, sort_by?: string) => void;
   filterItems?: {};
   sortSettings?: SortSetting[];
   autoSearch?: boolean;
+  initialValue?: {
+    query?: string;
+    filters?: {};
+    sort?: string;
+  };
 }) {
+  const defaultSort = sortSettings?.length ? sortSettings[0].value : "";
   const [keywords, setKeywords] = useState<string>("");
   const [filters, setFilters] = useState<any>({});
-  const [sort_by, setSortBy] = useState<string>("");
+  const [sort_by, setSortBy] = useState<string>(defaultSort);
 
   function handleSubmit(event: React.FormEvent<UsernameFormElement>) {
     event.preventDefault();
     onSubmitKeywords(keywords, filters, sort_by);
   }
 
-  function handleReset() {
+  function handleReset(event: React.FormEvent<UsernameFormElement>) {
```

Comment:
We already import the `ChangeEvent`, can we also move the `FormEvent` to the top level import?